Start work on projects

stefvanschie opened the pull request:
Reintroduced the PsiElementHolder since now both PsiLoopElement and
PsiFile use it. PsiFile is a bairnfuck file which automatically parse
the code into the PSI. A Directory is basically what the name suggests,
but it contains PsiFiles and Files as children instead of just Files
like the File class would have. A Project is a directory with a name.

Trigary commented:
The PsiElementHolder interface is no longer necessary for the PsiLoopElement to function (that's why it was removed). Reintroducing it breaks the PsiLoopElement's immutability without extending its functionality (as far as I can see, but it may prove useful in the future) since it already successfully performs its task.

stefvanschie commented:
Alright I removed the PsiElementHolder again.

stefvanschie committed:
Change usages of PsiElementFactory

Changed the usage of PsiElementFactory in this branch to match up with
the contents of the master branch.

stefvanschie committed:
Resolve merge conflict

stefvanschie committed:
Rebase merge conflict

stefvanschie commented:
Fixed merge conflict.
